src/server/kernel/wseat.h

```diff
@@ -105,6 +105,7 @@ class WAYLIB_SERVER_EXPORT WSeat : public WWrapObject, public WServerInterface
     void requestCursorShape(WAYLIB_SERVER_NAMESPACE::WGlobal::CursorShape shape);
     void requestCursorSurface(WAYLIB_SERVER_NAMESPACE::WSurface *surface, const QPoint &hotspot);
     void requestDrag(WAYLIB_SERVER_NAMESPACE::WSurface *surface);
+    void dragSurfaceDestroyed();
```

Comment:
This signal is not good, you can connect the WSurface::aboutToBeInvalidated to replace if care this surface if destroyed.